src/main/scala/is/hail/utils/IntIterator.scala

```diff
@@ -0,0 +1,14 @@
+package is.hail.utils
+
+abstract class IntIterator extends Iterator[Int] {
+  def nextInt(): Int
+  override def next(): Int = nextInt()
+
+  override def count(p: Int => Boolean): Int = {
```

Comment:
Ok, that makes sense, and I read more about type erasure and specialization:
http://www.scala-notes.org/2011/04/specializing-for-primitive-types/

I've hard-coded the function as follows:

```
  def countNonNegative(): Int = {
    var cnt = 0
    while (hasNext) {
      if (nextInt() >= 0) cnt += 1
    }
    cnt
  }
```